bundler/lib/bundler/index.rb

```diff
@@ -65,11 +63,14 @@ def search(query, base = nil)
     def unsorted_search(query, base)
       results = local_search(query, base)
 
-      seen = results.map(&:full_name).to_set unless @sources.empty?
+      seen = results.map(&:full_name).uniq unless @sources.empty?
 
       @sources.each do |source|
         source.unsorted_search(query, base).each do |spec|
-          results << spec if seen.add?(spec.full_name)
+          next if seen.include?(spec.full_name)
```

Comment:
is this a potential performance regression? would we be better off using a dictionary here to keep amortized constant-time inclusion checks?

Reply:
I'm not sure, I'll have a closer look.

The main motivation of this PR was that I found that in some cases using `Set` performs much worse than using plain Arrays for some reason (see the PR description), so I took the chance to remove `set` usages to fix that performance regression and at the same time support `set` as a gem.

Reply:
@segiddins I run a few tests to check performance and I didn't see any noticiable difference, I think it's because this is tipically a very small array.

I'd say let's go with what we have to allow `set` inside `Gemfile`. If this causes issues, they can be addressed later :+1:.